hack/test-go.sh

```diff
@@ -152,14 +187,18 @@ runTests() {
     fi
   fi
 
+  local junit_filename_prefix=$(junitFilenamePrefix)
+
   # If we're not collecting coverage, run all requested tests with one 'go test'
   # command, which is much faster.
   if [[ ! ${KUBE_COVER} =~ ^[yY]$ ]]; then
-    kube::log::status "Running unit tests without code coverage"
+    kube::log::status "Running tests without code coverage"
     go test "${goflags[@]:+${goflags[@]}}" \
       ${KUBE_RACE} ${KUBE_TIMEOUT} "${@+${@/#/${KUBE_GO_PACKAGE}/}}" \
-     "${testargs[@]:+${testargs[@]}}"
-    return 0
+     "${testargs[@]:+${testargs[@]}}" \
+     | tee ${junit_filename_prefix:+"${junit_filename_prefix}.stdout"} && rc=$? || rc=$?
+    produceJUnitXMLReport ${junit_filename_prefix}
+    return $rc
```

Comment:
`${rc}` for consistency (the file is strongly `${var}` consistent, which is also style guide preferred)